frontend/lib/src/AppNode.ts

```diff
@@ -223,6 +229,20 @@ export class ElementNode implements AppNode {
     throw new Error("'setIn' cannot be called on an ElementNode")
   }
 
+  public clearPageNodes(mainScriptHash: string): AppNode | undefined {
+    if (this.activeScriptHash !== mainScriptHash) {
+      return undefined
+    }
+
+    return new ElementNode(
+      this.element,
+      this.metadata,
+      this.scriptRunId,
+      this.activeScriptHash,
+      this.fragmentId
```

Comment:
It looks like a new `ElementNode` is created and returned, but with the existing `element`, `metadata` etc. So its a little bit unintuitive to me why this results in any clearing. Could you maybe add a comment explaining why this is clearing anything, please?